content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,108 @@ title="Specify the environment"
 weight=99
 +++
 
+Environment variables are a common way to configure various buildpacks at build-time.
+
 <!--more-->
 
-This page is a stub! The CNB project is applying to [Google Season of Docs](https://developers.google.com/season-of-docs/docs/timeline) to receive support for improving our documentation. Please check back soon.
+### POSIX Path Variables
+
+The following layer path environment variables MUST be set by the `lifecycle` during the `build` and `launch` phases in order to make buildpack dependencies accessible.
+
+During the `build` phase, each variable designated for build MUST contain absolute paths of all previous buildpacks’ `<layers>/<layer>/` directories that are designated for build.
```

Comment:
Okay let's start by saying that when the lifecycle runs buildpacks, it strips everything out of the environment that might have been defined on the build-time base image. It only allows a select set of pre-configured environment variables through: https://github.com/buildpacks/lifecycle/blob/a43d5993a4f2cc23c44b6480ba2ab09fe81d57ed/env/build.go#L9-L19

Then, it applies "buildpack defined" environment variables, which is anything that might have been configured by a buildpack that ran earlier during the build phase. Note that buildpacks can't set environment variables for other buildpacks during the detect phase.

If we look at this directory tree:
```
layers/
└── some-buildpack-id
    └── some-build-layer
        ├── bin
        │   └── some-binary
        ├── env
        │   └── SOME_FIRST_VAR
        ├── env.build
        │   └── SOME_SECOND_VAR
        └── lib
            ├── some-shared-library
            └── some-static-library
```

and the table here: https://github.com/buildpacks/spec/blob/main/buildpack.md#layer-paths

* `some-binary` is added to `PATH` for buildpacks that follow
* `some-shared-library` and `some-static-library` are added to `LD_LIBRARY_PATH` and `LIBRARY_PATH`
(I didn't illustrate the modification of `CPATH` and `PKG_CONFIG_PATH`, but hopefully this is enough to paint the picture)

`SOME_FIRST_VAR` takes precedence over `SOME_SECOND_VAR` when applying the environment modification rules, which we should discuss separately.

Reply:
Note that the above is only relevant for build because the layer "some-build-layer" is a build layer, meaning that some-build-layer.toml has

```
[types]
build = true
```